stakingModel/contracts/UBIScheme.sol

```diff
@@ -182,29 +203,41 @@ contract UBIScheme is AbstractUBI {
         // within the calculation.
         uint256 newDistribution = distributionFormula(0, account);
 
+        // pending user might claims tokens twice in the first day
+        if(pendingUserClaims[account].isEligible && pendingUserClaims[account].day == currentDay) {
+            PendingUser memory user = pendingUserClaims[account];
+            user.isEligible = false;
+            pendingUserClaims[account] = user;
+            GoodDollar token = GoodDollar(address(avatar.nativeToken()));
+            token.transfer(account, newDistribution);
+            emit UBIClaimed(account, newDistribution);
+            return true;
+        }
         // active user which has not claimed today yet, ie user last claimed < today
-        if(isRegistered(account) && !fishedUsersAddresses[account] &&
+        else if(isRegistered(account) && !fishedUsersAddresses[account] &&
             ((lastClaimed[account].sub(periodStart)) / 1 days) < currentDay) {
             lastClaimed[account] = now;
             claimDay[currentDay].hasClaimed[account] = true;
-            GoodDollar token = GoodDollar(address(avatar.nativeToken()));
-            token.transfer(account, newDistribution);
             Day memory day = claimDay[currentDay];
             day.amountOfClaimers = day.amountOfClaimers.add(1);
             day.claimAmount = day.claimAmount.add(newDistribution);
             claimDay[currentDay] = day;
+            GoodDollar token = GoodDollar(address(avatar.nativeToken()));
+            token.transfer(account, newDistribution);
             emit UBIClaimed(account, newDistribution);
             return true;
         }
         else if(!isRegistered(account) || fishedUsersAddresses[account]) { // a unregistered or fished user
             activeUsersCount = activeUsersCount.add(1);
+            prevDayActivatedCount = prevDayActivatedCount.add(1);
             fishedUsersAddresses[account] = false;
-            lastClaimed[account] = now; // marks last claimed as today
+            // marks last claimed as today
+            lastClaimed[account] = now;
+            // adds the user to the pending list
+            pendingUserAddresses.push(account);
```

Comment:
dont use push, use prevDayActivatedCount as index this way we can reuse array slots
each day it is being reset to 0, just need to save it before reset to another variable dailyDistributionCount